services/thermal_mgr/thermal_mgr.c

```diff
@@ -1,71 +1,125 @@
-  /* Send an event to the thermal manager queue */
-
-  return ERR_CODE_SUCCESS;
-}
-
-void osHandlerLM75BD(void) {
-  /* Implement this function */
-}
-
-static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
-  while (1) {
-    
-  }
-}
-
-void addTemperatureTelemetry(float tempC) {
-  printConsole("Temperature telemetry: %f deg C\n", tempC);
-}
-
-void overTemperatureDetected(void) {
-  printConsole("Over temperature detected!\n");
-}
-
-void safeOperatingConditions(void) { 
-  printConsole("Returned to safe operating conditions!\n");
-}
+#include "thermal_mgr.h"
+#include "errors.h"
+#include "lm75bd.h"
+#include "console.h"
+#include "logging.h"
+
+
+#include <FreeRTOS.h>
+#include <os_task.h>
+#include <os_queue.h>
+
+#include <string.h>
+
+#define THERMAL_MGR_STACK_SIZE 256U
+
+static TaskHandle_t thermalMgrTaskHandle;
+static StaticTask_t thermalMgrTaskBuffer;
+static StackType_t thermalMgrTaskStack[THERMAL_MGR_STACK_SIZE];
+
+#define THERMAL_MGR_QUEUE_LENGTH 10U
+#define THERMAL_MGR_QUEUE_ITEM_SIZE sizeof(thermal_mgr_event_t)
+
+static QueueHandle_t thermalMgrQueueHandle;
+static StaticQueue_t thermalMgrQueueBuffer;
+static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR_QUEUE_ITEM_SIZE];
+
+static void thermalMgr(void *pvParameters);
+
+void initThermalSystemManager(lm75bd_config_t *config) {
+  memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
+  memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
+  
+  thermalMgrTaskHandle = xTaskCreateStatic(
+    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+
+  memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
+  memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
+
+  thermalMgrQueueHandle = xQueueCreateStatic(
+    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
+
+}
+
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+  //first gather all edge cases where event does not work
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+  //check if the queue is initialized
+  else if (thermalMgrQueueHandle == NULL) {
+    printConsole("Thermal Manager queue not initialized.\n");
+    return ERR_CODE_INVALID_STATE;
+  }
+  //collect the xQueueSend in result object
+  BaseType_t result = xQueueSend(thermalMgrQueueHandle, event, 0);
+  if (result != pdTRUE) {
+    printConsole("Event could not be sent to Thermal Manager queue.\n");
+    return ERR_CODE_QUEUE_FULL;
+  }
+
+  return ERR_CODE_SUCCESS;
+}
+
+void osHandlerLM75BD(void) {
+  thermal_mgr_event_t event;
+  event.type = THERMAL_MGR_EVENT_OVER_TEMP_RECIEVED;
+  thermalMgrSendEvent(&event);
+}
+
+static void thermalMgr(void *pvParameters) {
+  thermal_mgr_event_t receivedEvent;
+  while (1) {
+    BaseType_t received = xQueueReceive(thermalMgrQueueHandle, &receivedEvent, 0);

```

Comment:
Make the block time portMAX_DELAY instead of 0 as the former will block the task until an event is received while 0 will to block at all and constantly poll the queue